tests/baselines/reference/user/enhanced-resolve.log

```diff
@@ -27,7 +27,9 @@ node_modules/enhanced-resolve/lib/CachedInputFileSystem.js(202,23): error TS2322
 node_modules/enhanced-resolve/lib/CachedInputFileSystem.js(207,27): error TS2322: Type 'null' is not assignable to type '(path: any) => any'.
 node_modules/enhanced-resolve/lib/CachedInputFileSystem.js(212,24): error TS2322: Type 'null' is not assignable to type '(path: any, callback: any) => void'.
 node_modules/enhanced-resolve/lib/CachedInputFileSystem.js(217,28): error TS2322: Type 'null' is not assignable to type '(path: any) => any'.
+node_modules/enhanced-resolve/lib/CachedInputFileSystem.js(221,14): error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
```

Comment:
these are correct as far as TS' current understanding of JS. We don't expect people to overwrite methods in the constructor:

```js
class C {
  constructor(fs) {
    this.readFile = fs.readFile ? fs.readFile : null
    if (this.readFile) {
      this._readJson = (path, callback) => this.readFile(...)
    }
  }
  readFile(path, callback) {
  }
}
```

This is from "CachedInputFileSystem.js", which tells you all need to know I think.